### PR TITLE
Create a ContainerResolver that returns items in priority order

### PR DESCRIPTION
**Pull request by ashmrtn**
Wrapper for existing graph.ContainerResolver structs that returns items
in priority order in the Items() function. Also has logic to filter out
containers if desired

This can help with things like preview backups where we only want to
backup a subset of items because it allows us to ensure we'll see the
containers we care about most first

---

#### Does this PR need a docs update or release note?

- [ ] :white_check_mark: Yes, it's included
- [ ] :clock1: Yes, but in a later PR
- [x] :no_entry: No

#### Type of change

- [x] :sunflower: Feature
- [ ] :bug: Bugfix
- [ ] :world_map: Documentation
- [ ] :robot: Supportability/Tests
- [ ] :computer: CI/Deployment
- [ ] :broom: Tech Debt/Cleanup

#### Test Plan

- [ ] :muscle: Manual
- [x] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: src/internal/m365/collection/exchange/container_resolver.go
Comment (pandeyabs) [inline src/internal/m365/collection/exchange/container_resolver.go:475]: Curious, does this need to be a failure? Am I correct in assuming that if we return a failure here, it would result in a preview backup failure?

File: src/internal/m365/collection/exchange/container_resolver.go
Comment (pandeyabs) [inline src/internal/m365/collection/exchange/container_resolver.go:426]:
```suggestion
	// resolvedInclude is the ordered list of resolved container IDs to add to the
```

File: src/internal/m365/collection/exchange/container_resolver.go
Comment (pandeyabs) [inline src/internal/m365/collection/exchange/container_resolver.go:534]: Nitpicking on a corner case. If there is a possible overlap between include & exclude lists, `Items()` code above would return the overlapping container as part of `res`. However, `ItemByID()` will return `nil`. 

Should we have the same behavior across both APIs? Another thing we can do is to catch overlaps in caller's unit or integration test.

File: src/internal/m365/collection/exchange/container_resolver_test.go
Comment (pandeyabs) [inline src/internal/m365/collection/exchange/container_resolver_test.go:293]: Curious, is it realistically possible to have `idNotInBase` in `includes` but not in base?

**Review by pandeyabs — COMMENTED** [review]
In general looks good! Have left a few comments & questions, can approve after that.
Let's wait for @ryanfkeepers's review as well.

**Comment by ryanfkeepers** [comment]
@ashmrtn I can get to this review as soon as the conflicts are fixed up.  Thanks!

File: src/internal/m365/collection/exchange/container_resolver.go
Comment (ryanfkeepers) [reply src/internal/m365/collection/exchange/container_resolver.go:475]: It's a good safety measure.  While graph api gives us good guarantees about the presence of an id, suppose we one day replace the inner item with an interface-compliant internal struct, or accidentally omit the ID from the item properties by selection.  This ensures we'll, at minimum, fail some integration tests.

Comment (ryanfkeepers) [reply src/internal/m365/collection/exchange/container_resolver.go:475]: nit: the error here is a little verbose
```suggestion
			return nil, clues.New("ranked include container missing ID").
```

File: src/internal/m365/collection/exchange/container_resolver.go
Comment (ryanfkeepers) [inline src/internal/m365/collection/exchange/container_resolver.go:482]: might want a comment explaining that we need to get the excluded containers in order to get their id for comparison.  Otherwise its not obvious why we're getting excluded items.

**Review by ryanfkeepers — APPROVED** [review]

File: src/internal/m365/collection/exchange/container_resolver.go
Comment (pandeyabs) [reply src/internal/m365/collection/exchange/container_resolver.go]: > This ensures we'll, at minimum, fail some integration tests.

Understood. My concern is around user facing errors, especially for preview backups. Wondering if it's better to show user some less data(say one less folder) vs showing them a failed backup. We can always catch errors using integ tests but treat them as ignorable errors in prod.

I understand failures would be mostly unlikely here, but wanted to have a UX discussion. Doesn't need to be addressed in this PR.

**Pull request merged**
